File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    # stack = []
    # for i in range(len(ingredient)):
    #     stack.append(ingredient[i])
    #     if stack[-4:] == [1, 2, 3, 1]:
    #         stack = stack[:-4]
    #         answer += 1
    
    s = []
    cnt = 0
    for i in ingredient:
        s.append(i)
        if s[-4:] == [1, 2, 3, 1]:
            cnt += 1
            for _ in range(4):
                s.pop()
    return cnt
        
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.java
import java.util.*;
class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        int idx = 0;
        int[] stack = new int[ingredient.length];
        
        for(int ingred : ingredient){
            stack[idx] = ingred;
            // System.out.println(Arrays.toString(stack));
            if(idx >= 3 
              && stack[idx] == 1
              && stack[idx -1] == 3
              && stack[idx -2] == 2
              && stack[idx -3] == 1){
                // System.out.println("pop!");
                answer++;
                idx -= 4;
            }
            
            idx++;
        }
        return answer;
    }
}